artifactory.py

```diff
@@ -1248,13 +1248,18 @@ def __new__(cls, *args, **kwargs):
 
         # Auth section
         apikey = kwargs.get("apikey")
+        token = kwargs.get("token")
         auth_type = kwargs.get("auth_type")
-        if apikey is None:
+
+        if apikey:
+            logging.debug("Use XJFrogApiAuth apikey")
+            obj.auth = XJFrogArtApiAuth(apikey=apikey)
```

Comment:
There're too many if - we have `if apikey \ if token` here and `if apikey \ if token` in `XJFrogArtApiAuth`.

Let's create a new class for TokenBased authentication in `auth.py`
```python
class XJFrogArtApiAuth:
    """Authenticaiton via ApiKey"""
    .... # Use X-JFrog-Art-Api header
class XJFrogArtBearerAuth:
   ... # Use Authorization Bearer header
```

In this case we'll have if only here when we choose one of authentication method.

Reply:
A new class `XJFrogArtBearerAuth` is added, could you please review it again?